### ISSUE TITLE
Error: Navigation took more than 50000ms

### ISSUE THREAD

**Issue by Debashis9012** [getgauge/taiko#428]
#### Excepted behaviour: 
If navigation happens within default time which is 30000ms then error should not throw.

#### Steps to replicate
Kindly run the below piece of code 

 openBrowser();
 goto("google.com");
 write("java concept of the day");
 click("Google Search");
click(text('https://javaconceptoftheday.com/'));

**Comment by NivedhaSenthil** [comment]
This is happening because there are few requests which are pending and not getting resolved. Providing option to wait for particular event on click should work for this usecase https://github.com/getgauge/taiko/issues/418.

**Comment by Apoorva-GA** [comment]
The same steps seems to work fine when ran as a script
```
const { openBrowser, goto, write, text, closeBrowser, click } = require('taiko');
(async () => {
    try {
        await openBrowser();
        await goto('google.com');
        await write('java concept of the day');
        await click('Google Search');
        await click(text('https://javaconceptoftheday.com/'));
    } catch (e) {
        console.error(e);
    } finally {
        await closeBrowser();
    }
})();
```

**Comment by NivedhaSenthil** [comment]
This issue is intermittent, as sometimes the pending request gets finished.

**Comment by Apoorva-GA** [comment]
Fix should be available in 26858ab.
Now we can wait for custom events on click and other APIs as well like,
`click(text('https://javaconceptoftheday.com/'), {'waitForEvents' : ['firstMeaningfulPaint']});`

**Comment by Debashis9012** [comment]
This is issue has been verified. To wait for specific events on click action we can define it by following the above comment.

Tested version: ``` 26858ab ```
